src/Header.js

```diff
@@ -1,7 +1,48 @@
 import "./Header.css";
+import filterIcon from "./filter.png";
 
-function Header() {
-    return <h2 className={"headerText"}>To Do List:</h2>
+function Header(props) {
+    return (
+        <div className={"headerFlexbox"}>
+            <p className={"headerText"}>To-Do List:</p>
+            <div className="dropdown">
+                <button className="dropdownButton"
+                        onClick={(e) => {
+                            props.onSortDropdownToggle(!props.showSortDropdown);
+                            e.stopPropagation();
+                            e.preventDefault();
+                        }}>
+                    <img className={"filterIcon"} src={filterIcon} alt={"filterIcon"}></img>
+                    </button>
+                {props.showSortDropdown && <div className={"paramDropdown"}>
+                    <span className={"sortDirection"}
+                       onClick={(e) => {
+                           props.onSortDirectionToggle();
+                           e.stopPropagation();
```

Comment:
Suggestion: have the stopPropagation be in an onClick on the dropdown itself.  That way, you don't have to repeat if for each of the elements of the dropdown.